python/metatensor-core/metatensor/io/_block.py

```diff
@@ -201,12 +202,8 @@ def _save_block(
         np.savez(file, **all_entries)
     else:
         lib = _get_library()
-        if isinstance(file, (str, pathlib.Path)):
-            if isinstance(file, str):
-                path = file.encode("utf8")
-            elif isinstance(file, pathlib.Path):
-                path = bytes(file)
-
+        if isinstance(file, str):
+            path = file.encode("utf8")
```

Comment:
So, I would rather keep the logic converting both `str` and Path to bytes, since on some platforms (linux mainly), a file name does not have to be utf-8. 

Then the code handling `.npz` extension can work on `bytes` instead of `str`

Reply:
Can you check what I did and let me know if you are OK with it? Thanks!

Reply:
There is still the same issue. The core of the problem is that there might be some `Path` objects which can NOT be transformed to `str` and/or can NOT have the resulting str utf8 encoded. So we need to work with `bytes` instead.